gcloud-java-pubsub/src/test/java/com/google/cloud/pubsub/ReceivedMessageTest.java

```diff
@@ -0,0 +1,188 @@
+    ReceivedMessage builtMessage = expectedMessage.toBuilder()
+        .payload("newPayload")
+        .id("newMessageId")
+        .attributes(attributes)
+        .publishTime(PUBLISH_TIME + 1)
+        .build();
+    assertSame(serviceMockReturnsOptions, builtMessage.pubsub());
+    assertEquals(SUBSCRIPTION, builtMessage.subscription());
+    assertEquals(ACK_ID, builtMessage.ackId());
+    assertEquals("newMessageId", builtMessage.id());
+    assertArrayEquals("newPayload".getBytes(Charsets.UTF_8), builtMessage.payload().toByteArray());
+    assertEquals("newPayload", builtMessage.payloadAsString());
+    assertEquals(attributes, builtMessage.attributes());
+    assertEquals(PUBLISH_TIME + 1, (long) builtMessage.publishTime());
+    builtMessage = builtMessage.toBuilder()
+        .payload(PAYLOAD)
+        .id(MESSAGE_ID)
+        .clearAttributes()
+        .addAttribute("key1", "value1")
+        .addAttribute("key2", "value2")
+        .publishTime(PUBLISH_TIME)
+        .build();
+    assertSame(serviceMockReturnsOptions, builtMessage.pubsub());
+    assertEquals(MESSAGE_ID, builtMessage.id());
+    assertEquals(PAYLOAD, builtMessage.payload());
+    assertEquals(PAYLOAD_STRING, builtMessage.payloadAsString());
+    assertEquals(ATTRIBUTES, builtMessage.attributes());
+    assertEquals(PUBLISH_TIME, builtMessage.publishTime());
+    compareReceivedMessage(expectedMessage, builtMessage);
+  }
+
+  @Test
+  public void testToBuilder() {
+    initializeExpectedMessage(2);
+    replay(pubsub);
+    compareReceivedMessage(expectedMessage, expectedMessage.toBuilder().build());
+  }
+
+  @Test
+  public void testAck() {
+    initializeExpectedMessage(1);
+    expect(pubsub.options()).andReturn(mockOptions);
+    pubsub.ack(SUBSCRIPTION, ACK_ID);
+    EasyMock.expectLastCall();
+    replay(pubsub);
+    initializeMessage();
+    message.ack();
+  }
+
+  @Test
+  public void testAckAsync() throws ExecutionException, InterruptedException {
+    initializeExpectedMessage(1);
+    expect(pubsub.options()).andReturn(mockOptions);
+    EasyMock.expect(pubsub.ackAsync(SUBSCRIPTION, ACK_ID))
+        .andReturn(Futures.<Void>immediateFuture(null));
+    EasyMock.expectLastCall();
+    replay(pubsub);
+    initializeMessage();
+    assertNull(message.ackAsync().get());
+  }
+
+  @Test
+  public void testModifyAckDeadline() {
+    initializeExpectedMessage(1);
+    expect(pubsub.options()).andReturn(mockOptions);
+    pubsub.modifyAckDeadline(SUBSCRIPTION, 10, TimeUnit.SECONDS, ACK_ID);
+    EasyMock.expectLastCall();
+    replay(pubsub);
+    initializeMessage();
+    message.modifyAckDeadline(10, TimeUnit.SECONDS);
+  }
+
+  @Test
+  public void testModifyAckDeadlineAsync() throws ExecutionException, InterruptedException {
+    initializeExpectedMessage(1);
+    expect(pubsub.options()).andReturn(mockOptions);
+    EasyMock.expect(pubsub.modifyAckDeadlineAsync(SUBSCRIPTION, 10, TimeUnit.SECONDS, ACK_ID))
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'EasyMock.expect' due to existing static import 'org.easymock.EasyMock.expect'](https://www.codacy.com/app/mziccard/gcloud-java/file/2671233162/issues/source?bid=3123109&fileBranchId=3332594#l169)